tools/flight_recorder/components/utils.py

```diff
@@ -364,7 +445,8 @@ def find_coalesced_group(
             break
 
     if len(found) > 1:
-        assert found[-1][1]["profiling_name"] == "nccl:coalesced"
+        if found[-1][1]["profiling_name"] != "nccl:coalesced":
+            logger.error("Rank %s does not have a coalesced end.", rank)
```

Comment:
We're changing this so we can throw errors on mismatched coalesced operations?

Reply:
Right. If we throw exception, then we won't run to the following analysis part.